Add Edit Secret step to Managing Secrets Using a Configuration File (#35384)
kubernetes/website: content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md

`YWRtaW5pc3RyYXRvcg==` decodes to `administrator`.

## Edit a Secret {#edit-secret}

> To edit the data in the Secret you created using a configuration
> file, modify the `data` or `stringData` field in your original configuration
> file and apply the file to your cluster. You can edit an existing `Secret` object unless it is [immutable](/docs/concepts/configuration/secret/#secret-immutable).

Reviewer: ```suggestion
To edit the data in the Secret you created using a manifest, modify either
the `data` or `stringData` field in your manifest, and apply the file to your cluster.
You can edit an existing `Secret` object unless it is [immutable](/docs/concepts/configuration/secret/#secret-immutable).
```

Author: Separately Tim, do you think we should rename the actual task to "Managing Secrets Using a Manifest"?

For example, if you want to change the password from the previous example to
`birdsarentreal`, do the following:

1. Encode the new password string:

   ```shell
   echo -n 'birdsarentreal' | base64
   ```

   The output is similar to:

   ```
   YmlyZHNhcmVudHJlYWw=
   ```

1. Update the `data` field with your new password string:

   ```yaml
   apiVersion: v1
   kind: Secret
   metadata:
     name: mysecret
   type: Opaque
   data:
     username: YWRtaW4=
     password: YmlyZHNhcmVudHJlYWw=
   ```

> 1. Apply the file to your cluster:

Reviewer: ```suggestion
1. Apply that manifest to your cluster:
```

   ```shell
   kubectl apply -f ./secret.yaml
   ```

   The output is similar to:

   ```
   secret/mysecret configured
   ```

> Kubernetes updates the existing `Secret` object, because you supplied a
> configuration file that had updated data with the same object name.

Reviewer: ```suggestion
Kubernetes updates the existing `Secret` object. In detail: the `kubectl` tool notices
that there is an existing object by the same name and so fetches that object,
plans changes, and submits the revised `Secret` to your cluster's control plane.

If you specified `kubectl apply --server-side` then the `kubectl` tool would
use [Server Side Apply](/docs/reference/using-api/server-side-apply/) instead.
```

> ## Delete a Secret {#clean-up}

Reviewer: ```suggestion
## Clean up
```

We have a convention for what this heading is named (in fact, as a separate concern, we could consider adding a localized string for it.

Author: So the reason I renamed it was because I saw it as a subtask, deleting a secret in a cluster, as opposed to cleaning up the resources created on this task page. I agree that the heading should normally be cleanup though

Author: (done in the latest push)
